rct229/rulesets/ashrae9012019/section19/section19rule2.py

```diff
@@ -0,0 +1,205 @@
+            ruleset_section_title="HVAC - General",
+            standard_section="Section G3.1.2.2",
+            is_primary_rule=True,
+            rmr_context="ruleset_model_instances/0",
+            list_path="$.fluid_loops[*]",
+        )
+
+    def create_data(self, context, data):
+        rmi_b = context.baseline
+
+        chiller_cooling_loop_b = find_all(
+            "$.buildings[*].chillers[*].cooling_loop", rmi_b
+        )
+
+        HW_fluid_loop_list = []
+        CW_fluid_loop_list = []
+        CHW_fluid_loop_list = []
+        for hvac_b in find_all(
+            "$.buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[*]",
+            rmi_b,
+        ):
+            hvac_id_b = hvac_b["id"]
+            if is_hvac_sys_heating_type_fluid_loop(
+                rmi_b, hvac_id_b
+            ) and is_hvac_sys_fluid_loop_attached_to_boiler(rmi_b, hvac_id_b):
+                HW_fluid_loop_list.append(
+                    getattr_(
+                        find_exactly_one_hvac_system(rmi_b, hvac_id_b),
+                        "HVAC",
+                        "heating_system",
+                        "hot_water_loop",
+                    )
+                )
+
+            if is_hvac_sys_preheating_type_fluid_loop(
+                rmi_b, hvac_id_b
+            ) and is_hvac_sys_preheat_fluid_loop_attached_to_boiler(rmi_b, hvac_id_b):
+                HW_fluid_loop_list.append(
+                    getattr_(
+                        find_exactly_one_hvac_system(rmi_b, hvac_id_b),
+                        "HVAC",
+                        "preheat_system",
+                        "hot_water_loop",
+                    )
+                )
+
+            if is_hvac_sys_cooling_type_fluid_loop(
+                rmi_b, hvac_id_b
+            ) and is_hvac_sys_fluid_loop_attached_to_chiller(rmi_b, hvac_id_b):
+                chilled_water_loop_b = getattr_(
+                    find_exactly_one_hvac_system(rmi_b, hvac_id_b),
+                    "HVAC",
+                    "cooling_system",
+                    "chilled_water_loop",
+                )
+                CHW_fluid_loop_list.append(chilled_water_loop_b)
+                if chilled_water_loop_b in chiller_cooling_loop_b:
+                    chiller_b = find_all_with_field_value(
+                        "$.chillers[*]",
+                        "cooling_loop",
+                        getattr_(
+                            hvac_b, "HVAC", "cooling_system", "chilled_water_loop"
+                        ),
+                        rmi_b,
+                    )
+                    if chiller_b.get("condensing_loop"):
+                        CW_fluid_loop_list.append(chiller_b["condensing_loop"])
+                else:
+                    primary_loop_b = find_exactly_one_child_loop(
+                        rmi_b, chilled_water_loop_b
+                    )
+                    CHW_fluid_loop_list.append(primary_loop_b["id"])
```

Comment:
I found this to be confusing.
line 96-100 is to get a loop ID from `cooling_system.chilled_water_loop`. If modeled correctly, this ID shall be a `child_loop` ID.
So line 102 append the `child_loop_id` to the `CHW_fluid_loop_list`

What is confusing to me is in the line 103-113. These lines are using the child loop to search for a condensing loop. The `child_loop` shall be a **Secondary Loop** associates with coils. The primary loop is connected to both chiller and secondary loop(s).
So, using child loop won't find a chiller, unless the model was incorrect.

In addition, I do not think you are searching for primary_loop_b as a child loop of a secondary loop. This should be another way around.